amazon-ecs-fargate-consul-connect-example-cdk-construct/lib/amazon-ecs-fargate-consul-connect-example-cdk-construct-stack.ts

```diff
@@ -0,0 +1,509 @@
+import * as cdk from '@aws-cdk/core';
+import * as ecr from '@aws-cdk/aws-ecr';
+import * as ecs from '@aws-cdk/aws-ecs';
+import { DockerImageAsset } from '@aws-cdk/aws-ecr-assets';
+import * as ecrdeploy from 'cdk-ecr-deployment';
+import * as awsLogs from '@aws-cdk/aws-logs';
+import * as ec2 from '@aws-cdk/aws-ec2'; 
+import * as iam from '@aws-cdk/aws-iam';
+
+export class AmazonEcsFargateConsulConnectExampleCdkConstructStack extends cdk.Stack {
+  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+//Import VPC where your consul connect server lives
+const vpc = ec2.Vpc.fromVpcAttributes(this, 'Consul-Connect-VPC', {
+  vpcId: '$CONSUL-CONNECT-VPC',
+  availabilityZones: ['$CONSUL-CONNECT-AZ'],
+  privateSubnetIds: ['$PRIVATE-SUBNETID'],
+  publicSubnetIds: ['$PUBLIC-SUBNETID']
+});
+
+//Import security group attached to the above VPC
+const importedSecurityGroup = ec2.SecurityGroup.fromSecurityGroupId(
+  this,
+  'imported-security-group',
+  '$SECURITY-GROUP',
+  {allowAllOutbound: true, mutable: true},
+);
+
+// Creating cluster where services will reside
+const cluster = new ecs.Cluster(this, "Consul-Connect-Cluster", {
+ vpc,
+});
+
+//Creating role for consul agent container in the service
+const consulAgentRole = new iam.Role(this, 'ConsulAgentRole', { assumedBy: new iam.ServicePrincipal("ecs-tasks.amazonaws.com") });
+
+//Creating task execution role
+const taskExecutionRole = new iam.Role(this, 'TaskExecutionRole', { assumedBy: new iam.ServicePrincipal("ecs-tasks.amazonaws.com") });
+
+//Adding policies to the consul agent role
+consulAgentRole.addToPolicy(
+  new iam.PolicyStatement({
+    effect: iam.Effect.ALLOW,
+    resources: ['*'],
+    actions: ['ec2:DescribeInstances', 'sts:AssumeRole'],
```

Comment:
why are we giving the agent role `sts:AssumeRole` permission? It shouldn't need to assume any role in order to work.

(ALSO: never never never grant `sts:AssumeRole` permission to resource `"*"`! This could potentially put other roles in your account at risk were this agent role to be compromised.)